src/metabase/api/database.clj

```diff
@@ -206,29 +206,25 @@
      filtered-dbs)))
 
 (defn- check-db-data-model-perms
-  [db include-editable-data-model?]
-  (if include-editable-data-model?
-    (let [filtered-dbs (filter-databases-by-data-model-perms [db])]
-      (when (seq filtered-dbs)
-        (first filtered-dbs)))
-    db))
-
-(defn- dbs-list [& {:keys [include-tables?
-                           include-saved-questions-db?
-                           include-saved-questions-tables?
-                           include-editable-data-model?
-                           exclude-uneditable-details?]}]
+  "Given a DB, checks that *current-user* has any data model editing perms for the DB. If yes, returns the DB,
+  with its tables also filtered by data model editing perms. If it does not, throws a permissions exception."
+  [db]
+  (let [filtered-dbs (filter-databases-by-data-model-perms [db])]
+    (api/check-403 (first filtered-dbs))))
```

Comment:
Changed this function to be consistent with the other "check" functions and throw with a 403 if the permission check fails.